Assets/QA/Tests/XRDeviceList/XRDeviceList.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using UnityEngine.UI;
+using UnityEngine.Experimental.Input;
+
+public class XRDeviceList : MonoBehaviour {
+
+    public Text deviceText;
+
+	// Use this for initialization
+	void Start () {
+        UpdateDeviceText();
+
+        InputSystem.onDeviceChange += OnDeviceChange;
+	}
+
+    void OnDeviceChange(InputDevice device, InputDeviceChange change)
+    {
+        UpdateDeviceText();
+    }
+
+    void UpdateDeviceText()
+    {
+        string deviceAccumulator = "";
+        for (int i = 0; i < InputSystem.devices.Count; i++)
+        {
+            deviceAccumulator += "<" + i + "> " + InputSystem.devices[i].displayName + "\n";
```

Comment:
`deviceAccumulator += string.Format("<{0}> {1}\n", i, InputSystem.devices[i].displayName);`
Could be easier to parse and edit later

Reply:
tomato tomato